daphne/src/lib.rs

```diff
@@ -573,6 +686,30 @@ impl DapTaskConfig {
 
         Ok(report_count >= self.min_batch_size)
     }
+
+    /// Leader: Resolve taskprov advertisement to send in a request to the Helper.
+    pub(crate) fn resolve_taskprove_advertisement(&self) -> Result<Option<String>, DapError> {
+        if let DapTaskConfigMethod::Taskprov { info: _ } = &self.method {
+            let encoded_taskprov_config = messages::taskprov::TaskConfig::try_from(self)?
+                .get_encoded_with_param(&self.version);
+            Ok(Some(encode_base64url(encoded_taskprov_config)))
+        } else if self.deprecated_taskprov && self.version != DapVersion::Draft02 {
```

Comment:
This if signals to me there `deprecated_taskprov` can only be `true` if the version is `Draft2`

Reply:
That's not true (see code comment below). The intent here is to abort loudly if we're in a situation where we can't proceed.

A simpler alternative would be to fail here if `self.version != DapVersion::Draft02`, regardless of the value of `self.deprecated_taskprov`.